pennylane/workflow/_capture_qnode.py

```diff
@@ -211,56 +208,54 @@ def _(*args, qnode, shots, device, execution_config, qfunc_jaxpr, n_consts, batc
                 temp_all_args.append(a[tuple(slices)])
             else:
                 temp_all_args.append(a)
-        temp_consts = temp_all_args[:n_consts]
-        temp_args = temp_all_args[n_consts:]
     else:
-        temp_consts = consts
-        temp_args = non_const_args
+        temp_all_args = args
 
     # Expand user transforms applied to the qfunc
+    accumulated_consts = []
     if getattr(qfunc_jaxpr.eqns[0].primitive, "prim_type", "") == "transform":
         transformed_func = qml.capture.expand_plxpr_transforms(
-            partial(qml.capture.eval_jaxpr, qfunc_jaxpr, temp_consts)
+            partial(qml.capture.eval_jaxpr, qfunc_jaxpr, [])
         )
 
-        qfunc_jaxpr = jax.make_jaxpr(transformed_func)(*temp_args)
-        temp_consts = qfunc_jaxpr.consts
-        qfunc_jaxpr = qfunc_jaxpr.jaxpr
+        qfunc_jaxpr = jax.make_jaxpr(transformed_func)(*temp_all_args)
+        accumulated_consts += qfunc_jaxpr.consts
+        qj = qfunc_jaxpr.jaxpr
+        qfunc_jaxpr = qj.replace(constvars=(), invars=qj.constvars + qj.invars)
 
     # Expand user transforms applied to the qnode
-    qfunc_jaxpr = qnode.transform_program(qfunc_jaxpr, temp_consts, *temp_args)
-    temp_consts = qfunc_jaxpr.consts
-    qfunc_jaxpr = qfunc_jaxpr.jaxpr
+    qfunc_jaxpr = qnode.transform_program(qfunc_jaxpr, [], *accumulated_consts, *temp_all_args)
+    accumulated_consts = qfunc_jaxpr.consts + accumulated_consts
+    qj = qfunc_jaxpr.jaxpr
+    qfunc_jaxpr = qj.replace(constvars=(), invars=qj.constvars + qj.invars)
 
     # Apply device preprocessing transforms
     graph_enabled = qml.decomposition.enabled_graph()
     try:
         qml.decomposition.disable_graph()
-        qfunc_jaxpr = device_program(qfunc_jaxpr, temp_consts, *temp_args)
+        qfunc_jaxpr = device_program(qfunc_jaxpr, [], *temp_all_args)
+        accumulated_consts = qfunc_jaxpr.consts + accumulated_consts
+        qj = qfunc_jaxpr.jaxpr
+        qfunc_jaxpr = qj.replace(constvars=(), invars=qj.constvars + qj.invars)
```

Comment:
I think it would be good to have a test for consts + multiple transforms (both qfunc and qnode) to make sure that the consts accumulation is correct.

Reply:
Shouldn't we already have some?

Reply:
I'm not sure if there are tests for consts with _multiple_ transforms. It's an edge case that should be explicitly tested in an integration test in my opinion.

Reply:
Ugk this is going to be one painful test to write... I'll give it my best shot.